apiserver/handler.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	err = config.ParseSpec()
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	err = api.registry.Post(key, config.ParsedSpec)
+	if err != nil {
+		response.AddHeader("Content-Type", "text/plain")
+		switch err.(type) {
+		case *model.ItemAlreadyExistsError:
+			api.writeError(http.StatusConflict, err.Error(), response)
+		default:
+			api.writeError(http.StatusInternalServerError, err.Error(), response)
+		}
+		return
+	}
+	if err = response.WriteHeaderAndEntity(http.StatusCreated, config); err != nil {
+		api.writeError(http.StatusInternalServerError, err.Error(), response)
+	}
+}
+
+// UpdateConfig updates the passed config object in the configuration registry
+func (api *API) UpdateConfig(request *restful.Request, response *restful.Response) {
+
+	// ToDo: check url name matches body name
+
+	params := request.PathParameters()
+	key, err := setup(params)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	config := &Config{}
+	err = request.ReadEntity(config)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	err = config.ParseSpec()
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	err = api.registry.Put(key, config.ParsedSpec)
+
+	if err != nil {
+		switch err.(type) {
+		case *model.ItemNotFoundError:
+			api.writeError(http.StatusNotFound, err.Error(), response)
+		default:
+			api.writeError(http.StatusInternalServerError, err.Error(), response)
+		}
+		return
+	}
+	if err = response.WriteHeaderAndEntity(http.StatusOK, config); err != nil {
+		api.writeError(http.StatusInternalServerError, err.Error(), response)
+	}
+}
+
+// DeleteConfig deletes the passed config object in the configuration registry
+func (api *API) DeleteConfig(request *restful.Request, response *restful.Response) {
+
+	params := request.PathParameters()
+	key, err := setup(params)
+	if err != nil {
+		api.writeError(http.StatusBadRequest, err.Error(), response)
+		return
+	}
+
+	err = api.registry.Delete(key)
+	if err != nil {
+		switch err.(type) {
+		case *model.ItemNotFoundError:
+			api.writeError(http.StatusNotFound, err.Error(), response)
+		default:
+			api.writeError(http.StatusInternalServerError, err.Error(), response)
+		}
+		return
+	}
+	response.WriteHeader(http.StatusOK)
+}
+
+// ListConfigs lists the configuration objects in the the configuration registry
+// If kind and namespace are passed then it retrieves all rules of a kind in a namespace
+// If kind is passed and namespace is an empty string it retrieves all rules of a kind across all namespaces
+func (api *API) ListConfigs(request *restful.Request, response *restful.Response) {
+
+	params := request.PathParameters()
+	namespace, kind := params["namespace"], params["kind"]
+
+	_, ok := model.IstioConfig[kind]
```

Comment:
`ok` assignment and check can be combined to minimize scope of `ok` variable. Also applies to line 216.